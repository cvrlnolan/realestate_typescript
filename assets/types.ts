export type Props = {
  page: number;
  isValid?: boolean;
  goNextPage: Function;
  goBack: Function;
};
