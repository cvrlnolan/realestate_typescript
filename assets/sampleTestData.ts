import faker from "faker";
import { categoryOptions } from "@/assets/categories";
import { countryOptions } from "@/assets/countries";

const estates = [
  //Data copied directly from MongoDB Atlas personal Cluster & Database
  {
    _id: faker.datatype.uuid(),
    title: faker.random.words(3),
    price: faker.finance.amount(100, 20000),
    status: Math.random() < 0.5 ? "sale" : "rent",
    address: faker.address.streetAddress(true),
    province: faker.address.state(),
    postal_code: faker.address.zipCode(),
    country:
      countryOptions[Math.floor(Math.random() * countryOptions.length)].value,
    category:
      categoryOptions[Math.floor(Math.random() * categoryOptions.length)].value,
    bedrooms: faker.datatype.number(10),
    baths: faker.datatype.number(5),
    surface_area: faker.datatype.number(10000),
    property_briefing: faker.lorem.sentences(5),
    additional_info: faker.lorem.sentences(5),
    cooling: Math.random() < 0.5,
    heating: Math.random() < 0.5,
    internet: Math.random() < 0.5,
    furniture: Math.random() < 0.5,
    parking: Math.random() < 0.5,
    email: faker.internet.email(),
    telephone: faker.phone.phoneNumber("+############"),
    createDate: new Date(),
    imgUrl:
      "https://firebasestorage.googleapis.com/v0/b/real-estate-2ff9c.appspot.com/o/Estates?alt=media&token=03b46643-93ed-4057-adf1-b8308cdaabbb",
    rating: faker.datatype.number(5),
    reviews: faker.datatype.number(100),
    totalRating: faker.datatype.number(100),
  },
  {
    _id: faker.datatype.uuid(),
    title: faker.random.words(3),
    price: faker.finance.amount(100, 20000),
    status: Math.random() < 0.5 ? "sale" : "rent",
    address: faker.address.streetAddress(true),
    province: faker.address.state(),
    postal_code: faker.address.zipCode(),
    country:
      countryOptions[Math.floor(Math.random() * countryOptions.length)].value,
    category:
      categoryOptions[Math.floor(Math.random() * categoryOptions.length)].value,
    bedrooms: faker.datatype.number(10),
    baths: faker.datatype.number(5),
    surface_area: faker.datatype.number(10000),
    property_briefing: faker.lorem.sentences(5),
    additional_info: faker.lorem.sentences(5),
    cooling: Math.random() < 0.5,
    heating: Math.random() < 0.5,
    internet: Math.random() < 0.5,
    furniture: Math.random() < 0.5,
    parking: Math.random() < 0.5,
    email: faker.internet.email(),
    telephone: faker.phone.phoneNumber("+############"),
    createDate: new Date(),
    imgUrl:
      "https://firebasestorage.googleapis.com/v0/b/real-estate-2ff9c.appspot.com/o/Estates?alt=media&token=03b46643-93ed-4057-adf1-b8308cdaabbb",
    rating: faker.datatype.number(5),
    reviews: faker.datatype.number(100),
    totalRating: faker.datatype.number(100),
  },
];

export { estates };
