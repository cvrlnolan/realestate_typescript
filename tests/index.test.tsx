/**
 * @jest-environment jsdom
 */

import React from "react";
import "@testing-library/jest-dom";
import { render, screen, waitFor } from "@testing-library/react";
import renderer from "react-test-renderer";
import Home from "../pages/index";
import { estates } from "@/assets/sampleTestData";

describe("Index Page", () => {
  it("should render correctly", async () => {
    render(<Home />);
    const image = await waitFor(() => screen.getByAltText("intro"));
    expect(image).toBeInTheDocument();
  });

  it("should match snapshot", () => {
    const tree = renderer.create(<Home />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
