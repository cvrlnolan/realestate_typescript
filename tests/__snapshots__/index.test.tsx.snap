// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index Page should match snapshot 1`] = `
Array [
  <div
    className="css-1hzqxck"
  >
    <div
      className="css-jfo7z"
    >
      <button
        aria-label="Open Menu"
        className="chakra-button css-1mh58fi"
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="chakra-icon css-onkibi"
          focusable={false}
          viewBox="0 0 24 24"
        >
          <path
            d="M 3 5 A 1.0001 1.0001 0 1 0 3 7 L 21 7 A 1.0001 1.0001 0 1 0 21 5 L 3 5 z M 3 11 A 1.0001 1.0001 0 1 0 3 13 L 21 13 A 1.0001 1.0001 0 1 0 21 11 L 3 11 z M 3 17 A 1.0001 1.0001 0 1 0 3 19 L 21 19 A 1.0001 1.0001 0 1 0 21 17 L 3 17 z"
            fill="currentColor"
          />
        </svg>
      </button>
      <div
        className="chakra-stack css-16awuyr"
      >
        <h2
          className="chakra-heading css-13mgebn"
          href="/"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          RealEstates
        </h2>
        <nav
          className="chakra-stack css-t14v28"
        >
          <a
            className="css-3xxwyh"
            href="/listings"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            Listings
          </a>
          <a
            className="css-3xxwyh"
            href="https://github.com/cvrlnolan/realestate_typescript"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            Repository
          </a>
        </nav>
      </div>
      <div
        className="css-70qvj9"
      >
        <button
          className="chakra-button css-1br927b"
          href="/estate/add"
          onClick={[Function]}
          onMouseEnter={[Function]}
          type="button"
        >
          <span
            className="chakra-button__icon css-1wh2kri"
          >
            <svg
              aria-hidden={true}
              className="chakra-icon css-onkibi"
              focusable={false}
              viewBox="0 0 24 24"
            >
              <path
                d="M0,12a1.5,1.5,0,0,0,1.5,1.5h8.75a.25.25,0,0,1,.25.25V22.5a1.5,1.5,0,0,0,3,0V13.75a.25.25,0,0,1,.25-.25H22.5a1.5,1.5,0,0,0,0-3H13.75a.25.25,0,0,1-.25-.25V1.5a1.5,1.5,0,0,0-3,0v8.75a.25.25,0,0,1-.25.25H1.5A1.5,1.5,0,0,0,0,12Z"
                fill="currentColor"
              />
            </svg>
          </span>
          Add Estate
        </button>
        <button
          aria-label="theme_button"
          className="chakra-button css-10a8y2i"
          data-testid="theme_button"
          type="button"
        >
          <svg
            aria-hidden={true}
            className="chakra-icon css-onkibi"
            focusable={false}
            viewBox="0 0 24 24"
          >
            <g
              fill="none"
              stroke="currentColor"
              strokeLinecap="round"
              strokeLinejoin="round"
              strokeWidth="2"
            >
              <circle
                cx="12"
                cy="12"
                r="5"
              />
              <path
                d="M12 1v2"
              />
              <path
                d="M12 21v2"
              />
              <path
                d="M4.22 4.22l1.42 1.42"
              />
              <path
                d="M18.36 18.36l1.42 1.42"
              />
              <path
                d="M1 12h2"
              />
              <path
                d="M21 12h2"
              />
              <path
                d="M4.22 19.78l1.42-1.42"
              />
              <path
                d="M18.36 5.64l1.42-1.42"
              />
            </g>
          </svg>
        </button>
      </div>
    </div>
  </div>,
  <div
    className="css-fxiqdw"
  >
    <div
      className="css-zge4aw"
    >
      <div
        className="css-hm1vf4"
      >
        <span
          style={
            Object {
              "background": "none",
              "border": 0,
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": "initial",
              "left": 0,
              "margin": 0,
              "opacity": 1,
              "overflow": "hidden",
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": "initial",
            }
          }
        >
          <img
            alt="intro"
            data-nimg="fill"
            decoding="async"
            sizes="100vw"
            src="/_next/image?url=%2Fimage.jpeg&w=3840&q=75"
            srcSet="/_next/image?url=%2Fimage.jpeg&w=640&q=75 640w, /_next/image?url=%2Fimage.jpeg&w=750&q=75 750w, /_next/image?url=%2Fimage.jpeg&w=828&q=75 828w, /_next/image?url=%2Fimage.jpeg&w=1080&q=75 1080w, /_next/image?url=%2Fimage.jpeg&w=1200&q=75 1200w, /_next/image?url=%2Fimage.jpeg&w=1920&q=75 1920w, /_next/image?url=%2Fimage.jpeg&w=2048&q=75 2048w, /_next/image?url=%2Fimage.jpeg&w=3840&q=75 3840w"
            style={
              Object {
                "border": "none",
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": 0,
                "left": 0,
                "margin": "auto",
                "maxHeight": "100%",
                "maxWidth": "100%",
                "minHeight": "100%",
                "minWidth": "100%",
                "objectFit": "cover",
                "objectPosition": undefined,
                "padding": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": 0,
              }
            }
          />
          <noscript>
            <img
              alt="intro"
              data-nimg="fill"
              decoding="async"
              loading="lazy"
              sizes="100vw"
              src="/_next/image?url=%2Fimage.jpeg&w=3840&q=75"
              srcSet="/_next/image?url=%2Fimage.jpeg&w=640&q=75 640w, /_next/image?url=%2Fimage.jpeg&w=750&q=75 750w, /_next/image?url=%2Fimage.jpeg&w=828&q=75 828w, /_next/image?url=%2Fimage.jpeg&w=1080&q=75 1080w, /_next/image?url=%2Fimage.jpeg&w=1200&q=75 1200w, /_next/image?url=%2Fimage.jpeg&w=1920&q=75 1920w, /_next/image?url=%2Fimage.jpeg&w=2048&q=75 2048w, /_next/image?url=%2Fimage.jpeg&w=3840&q=75 3840w"
              style={
                Object {
                  "border": "none",
                  "bottom": 0,
                  "boxSizing": "border-box",
                  "display": "block",
                  "height": 0,
                  "left": 0,
                  "margin": "auto",
                  "maxHeight": "100%",
                  "maxWidth": "100%",
                  "minHeight": "100%",
                  "minWidth": "100%",
                  "objectFit": "cover",
                  "objectPosition": undefined,
                  "padding": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "width": 0,
                }
              }
            />
          </noscript>
        </span>
        <div
          className="css-tu502g"
        >
          <div
            className="chakra-stack css-1phsyom"
          >
            <p
              className="chakra-text css-m0kk7z"
            >
              Welcome to RealEstates
            </p>
            <p
              className="chakra-text css-sw4nx8"
            >
              Visit, buy or rent.
            </p>
            <button
              className="chakra-button css-5c2176"
              href="/listings"
              onClick={[Function]}
              onMouseEnter={[Function]}
              type="button"
            >
              Show Listings
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      className="css-1ms4blg"
    >
      <p
        className="chakra-text css-48diyc"
      >
        Developed by
        <p
          className="chakra-text css-q908ka"
          href="https://carlnolan.lootyclub.com"
          onClick={[Function]}
          onMouseEnter={[Function]}
        >
          Carl Nolan
        </p>
      </p>
    </div>
  </div>,
]
`;
